### DISCUSSION TITLE
fetch endpoint list with last_login_time

### DISCUSSION THREAD

**Discussion by sagar1396** [CrowdStrike/falconpy#820] [Q&A] [answered]
how to fetch all the endpoints from crowdstrike using falconpy
how to fetch active endpoints from crowdstrike using falconpy

**Comment by jshcodes** [comment, marked as answer]
Hi @sagar1396 -

For querying a list of all endpoints within your Falcon tenant, you should check out the [Sensor versions by Hostname](https://github.com/CrowdStrike/falconpy/blob/main/samples/hosts/sensor_versions_by_hostname_scrolling.py) sample.

Building on that example, you could provide a `filter` to the [QueryDevicesByFilterScroll](https://www.falconpy.io/Service-Collections/Hosts.html#querydevicesbyfilterscroll) operation ([line 50](https://github.com/CrowdStrike/falconpy/blob/main/samples/hosts/sensor_versions_by_hostname_scrolling.py#L50) of this sample) to further limit the returned results.  From your question, I think you may be interested in the [`last_seen`](https://www.falconpy.io/Service-Collections/Hosts.html#available-filters-1) property.

For more details on FQL (Falcon Query Language) check [this page](https://www.falconpy.io/Usage/Falcon-Query-Language.html).

Thank you for the question, let us know if you run into any more! 😄
